Bottom: 4c42e349217173baee2dd6659d316dafe506bede
Top:    171c1ac8b619cd500955a08dbc94f5ebc6653eb2
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-10-02 14:58:41 +0800

Refresh of m2

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 79447e3b..6f78cfa 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1171,7 +1171,7 @@ static int tun_xdp_xmit(struct net_device *dev, struct xdp_buff *xdp)
 	struct tun_struct *tun = netdev_priv(dev);
 	struct tun_file *tfile = tun->tfiles[0];
 	struct ptr_ring *ring = &tfile->xdp_ring;
-	struct xdp_buff *buff = xdp->data;
+	struct xdp_buff *buff = xdp->data_hard_start;
 	int headroom = xdp->data - xdp->data_hard_start;
 
 	/* Assure headroom is available for storing xdp */
